src/include/daos_types.h

```diff
@@ -31,6 +31,12 @@
 extern "C" {
 #endif
 
+#if __GNUC__ >= 4
+	#define DAOS_API __attribute__ ((visibility ("default")))
```

Comment:
(style)  space prohibited between function name and open parenthesis '('
(style)  Macros with complex values should be enclosed in parentheses